day18/main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	aoc "github.com/golang-insiders/advent-of-code-2023/library"
+)
+
+func main() {
+	if len(os.Args) != 2 {
+		fmt.Println("input file path not provided")
+		return
+	}
+	input := aoc.ReadFileLineByLine(os.Args[1])
+
+	fmt.Println("answer for part 1: ", sol1(input))
+	fmt.Println("answer for part 2: ", sol2(input))
+}
+
+type node struct {
+	x int64
+	y int64
+}
+
+func sol1(input []string) int64 {
+	start := node{0, 0}
+	vertices := []node{}
```

Comment:
If I may, there's almost no good reason to initialize a slice this way: `v := []T{}`.

In this very case, it would be better to use make with a given capacity or length. 

But in the case where you don't know the future length, using `var v []T` is more appropriate. It creates an empty *and* nil slice. Your code creates an empty but non-nil slice. It means that in some cases, if your slice stays empty, you'll do an alloc for nothing.

I know it can sound pretty nitpicking but in general, we shouldn't do this (look at the Go repo, I'm pretty sure there's not a single slice initialized this way) :)